src/server_manager/web_app/cloud_accounts.ts

```diff
@@ -33,18 +50,29 @@ export class CloudAccounts {
   }
 
   getDigitalOceanAccount(): Account {
-    const token = this.getTokenFromStorage();
-    if (token) {
-      return this.digitalOceanAccountFactory(token);
+    const persistedAccount = this.readDigitalOceanAccount();
+    if (persistedAccount) {
+      const accountId = this.makeUniqueAccountId(persistedAccount.id, CloudId.DigitalOcean);
+      return this.digitalOceanAccountFactory(
+          accountId, persistedAccount.name, persistedAccount.credential);
     }
     return null;
   }
 
-  private writeTokenToStorage(token: string): void {
-    this.storage.setItem(this.DIGITALOCEAN_TOKEN_STORAGE_KEY, token);
+  private writeDigitalOceanAccount(persistedAccount: PersistedDigitalOceanAccount): void {
+    this.storage.setItem(this.DIGITALOCEAN_TOKEN_STORAGE_KEY, JSON.stringify(persistedAccount));
+  }
+
+  private readDigitalOceanAccount(): PersistedDigitalOceanAccount {
+    let result = null;
+    const data = this.storage.getItem(this.DIGITALOCEAN_TOKEN_STORAGE_KEY);
+    if (data) {
+      result = JSON.parse(data);
+    }
+    return result;
   }
 
-  private getTokenFromStorage(): string {
-    return this.storage.getItem(this.DIGITALOCEAN_TOKEN_STORAGE_KEY);
+  private makeUniqueAccountId(cloudSpecificAccountId: string, cloudId: CloudId): string {
+    return `${cloudId}#${cloudSpecificAccountId}`;
```

Comment:
See my comment on globally unique ids. I think that can live in the application code instead. We never write the global ids to storage, so its format doesn't matter much, since we can easily change it. Perhaps a string would be easier to debug. Perhaps a hash would make it clearer it's supposed to be treated as an opaque string.